In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [33]:
df1 = pd.read_csv('cleaned_trans_df.csv')

In [35]:
df1.head()

Date  STORE_NBR  LYLTY_CARD_NBR  TXN_ID  PROD_NBR  \
0  5/20/2019          1            1343     383        61   
1  5/16/2019          4            4149    3333        16   
2  5/17/2019          7            7215    7176        16   
3  5/15/2019         19           19272   16686        44   
4  5/18/2019         45           45220   41651        22   

                          product_name Brand_name  packet_size  prod_qty  \
0     Smiths Crinkle Cut Chips Chicken     Smiths          170         2   
1  Smiths Crinkle Chips Salt & Vinegar     Smiths          330         1   
2  Smiths Crinkle Chips Salt & Vinegar     Smiths          330         1   
3             Thins Chips Light& Tangy      Thins          175         1   
4            Thins Chips Originl saltd      Thins          175         1   

   total_sales  Year  month_num month_name    weekday  weekday_num  quarter  
0          3.1  2019          5        May     Monday            2        2  
1          5.7  2019          5        May   Thursday            5        2  
2          5.7  2019          5        May     Friday            6        2  
3          3.3  2019          5        May  Wednesday            4        2  
4          3.3  2019          5        May   Saturday            7        2

In [37]:
df2 = pd.read_csv('purchase_behaviour.csv')

In [39]:
df2.head()

LYLTY_CARD_NBR               lifestage premium_customer
0            1000   YOUNG SINGLES/COUPLES          Premium
1            1002   YOUNG SINGLES/COUPLES       Mainstream
2            1003          YOUNG FAMILIES           Budget
3            1004   OLDER SINGLES/COUPLES       Mainstream
4            1005  MIDAGE SINGLES/COUPLES       Mainstream

In [41]:
display(df1.shape)
display(df2.shape)

(49770, 16)

(72637, 3)

### Merging both datasets

Our cleaned transaction dataset has around 49,000 rows while the customer dataset has around 72,637 rows. So it is clear that we want all the data from the cleaned transaction dataset and the matching data from the customer dataset (left join). It can be done by using pandas library merge function and since we can see that in both datasets `LYLTY_CARD_NBR` column exists which will be a primary key and both datasets will be merged(left join) on the basis of this key

In [46]:
# merge both datasets
combined_df = pd.merge(df1, df2, how = 'left', on = 'LYLTY_CARD_NBR')

In [48]:
combined_df.head()

Date  STORE_NBR  LYLTY_CARD_NBR  TXN_ID  PROD_NBR  \
0  5/20/2019          1            1343     383        61   
1  5/16/2019          4            4149    3333        16   
2  5/17/2019          7            7215    7176        16   
3  5/15/2019         19           19272   16686        44   
4  5/18/2019         45           45220   41651        22   

                          product_name Brand_name  packet_size  prod_qty  \
0     Smiths Crinkle Cut Chips Chicken     Smiths          170         2   
1  Smiths Crinkle Chips Salt & Vinegar     Smiths          330         1   
2  Smiths Crinkle Chips Salt & Vinegar     Smiths          330         1   
3             Thins Chips Light& Tangy      Thins          175         1   
4            Thins Chips Originl saltd      Thins          175         1   

   total_sales  Year  month_num month_name    weekday  weekday_num  quarter  \
0          3.1  2019          5        May     Monday            2        2   
1          5.7  2019          5        May   Thursday            5        2   
2          5.7  2019          5        May     Friday            6        2   
3          3.3  2019          5        May  Wednesday            4        2   
4          3.3  2019          5        May   Saturday            7        2   

                lifestage premium_customer  
0  MIDAGE SINGLES/COUPLES           Budget  
1  MIDAGE SINGLES/COUPLES           Budget  
2  MIDAGE SINGLES/COUPLES           Budget  
3  MIDAGE SINGLES/COUPLES           Budget  
4  MIDAGE SINGLES/COUPLES           Budget

Both datasets have been merged together.

In [53]:
combined_df.shape

(49770, 18)

In [55]:
combined_df.isnull().sum()

Date                0
STORE_NBR           0
LYLTY_CARD_NBR      0
TXN_ID              0
PROD_NBR            0
product_name        0
Brand_name          0
packet_size         0
prod_qty            0
total_sales         0
Year                0
month_num           0
month_name          0
weekday             0
weekday_num         0
quarter             0
lifestage           0
premium_customer    0
dtype: int64

There are no null values which is good

In [60]:
# checking for duplicated values
combined_df.duplicated().sum()

0

We will be conducting further data analysis on this dataset to explore more about the data and to understand the customer segmentations so let's save this file into excel file.

In [67]:
combined_df.to_csv('Final_dataset.csv')